controlplane/kubeadm/internal/workload_cluster_coredns.go

```diff
@@ -251,6 +252,7 @@ func (w *Workload) updateCoreDNSCorefile(ctx context.Context, info *coreDNSInfo)
 
 	// Patching the coredns deployment to point to the Corefile-backup
 	// contents before performing the migration.
+	// NOTE: This object doesn't implement observedGeneration. No need to add the WithStatusObservedGeneration option.
```

Comment:
This comment seems a bit misleading, we shouldn't be updating observedGeneration because we are not managing this resource, just trying to get information from an externally reconciled resource.